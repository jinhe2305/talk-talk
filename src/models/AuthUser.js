import { User } from './User';

export class AuthUser extends User {
  pushToken: string = '';

  constructor(pushToken?: string) {
    super();
    this.pushToken = pushToken;
  }
}
