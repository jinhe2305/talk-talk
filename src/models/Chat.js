import { User } from './User';

export class Chat {
  id: string;
  sender: User = new User();
  message: String = '';
  created: Date = Date(0);
  updated: Date = Date(0);

  constructor(id: string, sender?: User, message?: string, created?: Date, updated?: Date) {
    this.id = id;
    this.sender = sender;
    this.message = message;
    this.created = created;
    this.updated = updated;
  }
}
