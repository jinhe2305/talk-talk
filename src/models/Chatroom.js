import { Chat } from './Chat';
import { User } from './User';

export class Chatroom {
  id: string;
  lastChat: Chat = new Chat();
  lastChatCnt: number = 0;
  chats: Chat[] = [];
  users: User[] = [];
  created: Date = Date(0);
  updated: Date = Date(0);

  constructor(
    id?: string,
    lastChat?: Chat,
    lastChatCnt: number,
    chats?: Chat[],
    users?: User[],
    created?: Date,
    updated?: Date
  ) {
    this.id = id;
    this.lastChat = lastChat;
    this.lastChatCnt = lastChatCnt;
    this.chats = chats;
    this.users = users;
    this.created = created;
    this.updated = updated;
  }
}
