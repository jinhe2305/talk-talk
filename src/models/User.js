import { Chatroom } from './Chatroom';

export class User {
  uid: string = '';
  displayName: string = '';
  photoURL: string = '';
  statusMsg: string = '';
  isOnline: boolean = false;
  friends: User[] = [];
  Chatrooms: Chatroom[] = [];
  created: Date = Date(0);
  updated: Date = Date(0);

  constructor(
    uid?: string,
    displayName?: string,
    photoURL?: string,
    statusMsg?: string,
    isOnline?: boolean,
    friends?: User[],
    chatrooms?: Chatroom[],
    created?: Date,
    updated?: Date,
  ) {
    this.uid = uid;
    this.displayName = displayName;
    this.photoURL = photoURL;
    this.statusMsg = statusMsg;
    this.isOnline = isOnline;
    this.friends = friends;
    this.chatrooms = chatrooms;
    this.created = created;
    this.updated = updated;
  }
}
