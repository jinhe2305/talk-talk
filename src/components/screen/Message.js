// @flow
import React, { Component } from 'react';
import {
  StyleSheet,
  TouchableOpacity,
  Image,
  Text,
  View,
  FlatList,
} from 'react-native';

import type {
  ____ViewStyleProp_Internal as ViewStyle,
  ____TextStyleProp_Internal as TextStyle,
  ____ImageStyleProp_Internal as ImageStyle,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes';

import EmptyListItem from '../shared/EmptyListItem';
import ChatroomListItem from '../shared/ChatroomListItem';
import { ratio, colors } from '../../utils/Styles';
import { getString } from '../../../STRINGS';
import { Chatroom } from '../../models/Chatroom';
import { Chat } from '../../models/Chat';
import { User } from '../../models/User';

type Styles = {
  container: ViewStyle,
}

const styles: Styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'transparent',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
  },
});

type Props = {
  navigation: any,
};
type State = {
  rooms: Chatroom[],
};

class Screen extends Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      rooms: [
        new Chatroom(
          '',
          new Chat(
            '',
            new User('sender_uid1', 'dooboolab', '', 'I am fine', true),
            'How are you?',
          ),
          6,
        ),
        new Chatroom(
          '',
          new Chat(
            '',
            new User('sender_uid2', 'Byun8585', '', 'hello folks', false),
            'Hi. This is student from react-native-seoul. Nice to meet you.',
          ),
          0,
        ),
      ],
    };
  }

  render() {
    return (
      <View style={styles.container}>
        <FlatList
          style={{
            alignSelf: 'stretch',
          }}
          contentContainerStyle={
            this.state.rooms.length === 0
              ? {
                flex: 1,
                alignSelf: 'stretch',
                alignItems: 'center',
                justifyContent: 'center',
              }
              : null
          }
          keyExtractor={(item, index) => index.toString()}
          data={this.state.rooms}
          renderItem={({ item }) => this.renderItem(item)}
          ListEmptyComponent={<EmptyListItem>{getString('NO_CONTENT')}</EmptyListItem>}
        />
      </View>
    );
  }

  renderItem = (item: Chatroom) => {
    return (
      <ChatroomListItem
        item={item}
        onPress={() => this.onItemClick(item.id)}
      />
    );
  }

  onItemClick = (itemId: string) => {
    this.props.navigation.navigate('Chat', { chatId: itemId });
  }
}

export default Screen;
