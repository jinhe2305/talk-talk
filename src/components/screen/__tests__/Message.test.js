import 'react-native';
import * as React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { shallow, render } from 'enzyme';

import Message from '../Message';
import ChatroomListItem from '../../shared/ChatroomListItem';
import { Chatroom } from '../../../models/Chatroom';
import { Chat } from '../../../models/Chat';
import { User } from '../../../models/User';

describe('rendering test', () => {
  let wrapper;
  let props;
  beforeEach(() => {
    props = {
      navigation: {
        navigate: jest.fn(),
      },
    };
    wrapper = shallow(<Message {...props} />);
  });
  it('should render as expected', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should render item', () => {
    const item = new Chatroom(
      '',
      new Chat(
        '',
        new User('sender_uid1', 'dooboolab', '', 'I am fine', true),
        'How are you?',
      ),
      6,
    );
    const rendered = wrapper.instance().renderItem(item);
    expect(wrapper.find(ChatroomListItem).last()).toHaveLength(1);
    // const shallowRendered = shallow(rendered);
    // shallowRendered.instance().props.onPress('aaaaaa');
  });
  it('should navigate on item click', () => {
    wrapper.instance().onItemClick('aaaaaa');
    expect(props.navigation.navigate).toHaveBeenCalledWith('Chat', {"chatId": "aaaaaa"});
  });
});
