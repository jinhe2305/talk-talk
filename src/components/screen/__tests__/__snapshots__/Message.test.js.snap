// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering test should render as expected 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <FlatList
    ListEmptyComponent={
      <Shared
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "flex": 1,
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        No content
      </Shared>
    }
    contentContainerStyle={null}
    data={
      Array [
        Chatroom {
          "chats": undefined,
          "created": undefined,
          "id": "",
          "lastChat": Chat {
            "created": undefined,
            "id": "",
            "message": "How are you?",
            "sender": User {
              "Chatrooms": Array [],
              "chatrooms": undefined,
              "created": undefined,
              "displayName": "dooboolab",
              "friends": undefined,
              "isOnline": true,
              "photoURL": "",
              "statusMsg": "I am fine",
              "uid": "sender_uid1",
              "updated": undefined,
            },
            "updated": undefined,
          },
          "lastChatCnt": 6,
          "updated": undefined,
          "users": undefined,
        },
        Chatroom {
          "chats": undefined,
          "created": undefined,
          "id": "",
          "lastChat": Chat {
            "created": undefined,
            "id": "",
            "message": "Hi. This is student from react-native-seoul. Nice to meet you.",
            "sender": User {
              "Chatrooms": Array [],
              "chatrooms": undefined,
              "created": undefined,
              "displayName": "Byun8585",
              "friends": undefined,
              "isOnline": false,
              "photoURL": "",
              "statusMsg": "hello folks",
              "uid": "sender_uid2",
              "updated": undefined,
            },
            "updated": undefined,
          },
          "lastChatCnt": 0,
          "updated": undefined,
          "users": undefined,
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    style={
      Object {
        "alignSelf": "stretch",
      }
    }
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</View>
`;
